### PR TITLE
Fix bugs in heapster core.

### PR DESCRIPTION
**Pull request by vishh**

### CODE REVIEW

File: sources/kube.go [hunk 1]
```diff
@@ -147,11 +147,11 @@ func (self *kubeSource) getPodInfo(nodeList *nodes.NodeList) ([]api.Pod, error)
 	var (
 		wg sync.WaitGroup
 	)
-	for _, pod := range pods {
+	for index := range pods {
```
Comment (vmarmol) [inline sources/kube.go]: index is unused

Comment (vmarmol) [inline sources/kube.go]: Ah I see now, this GitHub diff is terrible :P

File: sources/kube.go [hunk 2]
```diff
@@ -147,11 +147,11 @@ func (self *kubeSource) getPodInfo(nodeList *nodes.NodeList) ([]api.Pod, error)
...
 		wg.Add(1)
 		go func(pod *api.Pod) {
 			defer wg.Done()
-			for _, container := range pod.Containers {
+			for index, container := range pod.Containers {
```
Comment (vmarmol) [inline sources/kube.go]: Lets not use container and only index?

File: sources/kube.go [hunk 3]
```diff
@@ -147,11 +147,11 @@ func (self *kubeSource) getPodInfo(nodeList *nodes.NodeList) ([]api.Pod, error)
...
 				rawContainer, err := self.getStatsFromKubelet(pod, container.Name)
 				if err != nil {
 					// Containers could be in the process of being setup or restarting while the pod is alive.
```
Comment (vmarmol) [inline sources/kube.go]: Is the log line below right? It will display "<container name>/<namespace> <pod>"

File: sources/pods.go [hunk 4]
```diff
@@ -81,9 +81,9 @@ func (self *realPodsApi) parsePod(podNodePair *podNodePair) *api.Pod {
 
 func (self *realPodsApi) parseAllPods(podNodePairs []podNodePair) []api.Pod {
 	out := make([]api.Pod, 0)
-	for _, podNodePair := range podNodePairs {
+	for i, podNodePair := range podNodePairs {
```
Comment (vmarmol) [inline sources/pods.go]: Should we use both for this file too? It seems confusing and error prone.

**Comment by vmarmol** [comment]
Can we add more description as to the bugs in the commit message? :)

File: sources/kube.go (hunk 2 above)
Comment (vishh) [inline sources/kube.go]: Having container results in lesser code inside the loop. :)

File: sources/kube.go (hunk 3 above)
Comment (vishh) [inline sources/kube.go]: Which log line? The two ones below seem to be fine.

File: sources/pods.go (hunk 4 above)
Comment (vishh) [inline sources/pods.go]: Agreed! Done.

File: sources/kube.go (hunk 1 above)
Comment (vishh) [inline sources/kube.go]: yeah. It is indeed confusing.

**Comment by vmarmol** [comment]
LGTM

**Pull request merged**